src/arsort.py

```diff
@@ -663,3 +666,84 @@ def make_dirs(argflag, fitsdict, filesort):
     # Return the name of the science targets
     return sci_targs
 
+
+def calib_setup(sciexp, sc, det, fitsdict, calib_dict,
+                write=False):
+    """ Define calibration setup
+    Parameters
+    ----------
+    sciexp
+    calib_dict
+    Returns
+    -------
+    """
+    import json, io
+    # Arc
+    idx = sciexp._spect['arc']['index'][sc]
+    disp_name = fitsdict["disperser"][idx[0]]
+    disp_angle = fitsdict["cdangle"][idx[0]]
+    # Common
+    dichroic = fitsdict["dichroic"][idx[0]]
+    decker = fitsdict["decker"][idx[0]]
+    slitwid = fitsdict["slitwid"][idx[0]]
+    slitlen = fitsdict["slitlen"][idx[0]]
+    # Detector
+    binning = fitsdict["binning"][idx[0]]
+    naxis0 = fitsdict["naxis0"][idx[0]]
+    naxis1 = fitsdict["naxis1"][idx[0]]
+
+    # Generate
+    # Don't nest deeper than 1
+    cdict = dict(disperser={'name': disp_name,
+                            'angle': disp_angle},
+                 dichroic=dichroic,
+                 slit={'decker': decker,
+                       'slitwid': slitwid,
+                       'slitlen': slitlen},
+                 detector={'binning': binning,
+                           'det': det,
+                           'naxis0': naxis0,
+                           'naxis1': naxis1},
+                 )
+
+    if len(calib_dict) == 0: # Generate
+        setup = 'A'
```

Comment:
I'm not opposed to using letters, but if there's more than 26 setups (!!) than this might be buggy. Might be better to use an int (expressed as a string) instead, i.e. setup = '1'.

We could then remove the "import string" above, too.